py/server/tests/test_arrow.py

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+import unittest
+
+import numpy as np
+import pyarrow.parquet as papq
+
+from deephaven import arrow as dharrow, dtypes, new_table, DHError, time_table
+from deephaven.column import byte_col, char_col, short_col, int_col, long_col, float_col, double_col, \
+    string_col, datetime_col
+from deephaven.table import Table
+from tests.testbase import BaseTestCase
+
+
+class ArrowTestCase(BaseTestCase):
+    test_table: Table
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cols = [
+            # bool_col(name="Boolean", data=[True, False]),
+            byte_col(name="Byte", data=(1, -1)),
+            char_col(name="Char", data='-1'),
+            short_col(name="Short", data=[1, -1]),
+            int_col(name="Int", data=[1, -1]),
+            long_col(name="Long", data=[1, -1]),
+            long_col(name="NPLong", data=np.array([1, -1], dtype=np.int8)),
+            float_col(name="Float", data=[1.01, -1.01]),
+            double_col(name="Double", data=[1.01, -1.01]),
+            string_col(name="String", data=["foo", "bar"]),
+            datetime_col(name="Datetime", data=[dtypes.DateTime(1), dtypes.DateTime(-1)]),
+        ]
+        cls.test_table = new_table(cols=cols)
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        del cls.test_table
+
+    def test_round_trip(self):
+        arrow_table = papq.read_table("tests/data/day_trades.parquet")
+
+        dh_table = dharrow.to_table(arrow_table)
+        pa_table = dharrow.to_arrow(dh_table)
+        dh_table_rt = dharrow.to_table(pa_table)
+        self.assert_table_equals(dh_table, dh_table_rt)
```

Comment:
This test feels like a land mine. If `to_table` creates an empty table then this test will pass, but would clearly be incorrect.

Could we compare the converted table to the version that our ParquetTools reader would generate instead?

Reply:
Good point, added another assert on table size to tighten the test a bit.

Reply:
Also added a test to specifically check against the table read in through DH's parquet tool